File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    // Function to find the maximum number of meetings that can
    // be performed in a meeting room.
    static bool compr(pair<int,int> a, pair<int,int> b){
        if(a.second == b.second){
            return a.first < b.first;
        }
        return a.second<b.second;
    }
    int maxMeetings(vector<int>& start, vector<int>& end) {
        // Your code here
        vector<pair<int,int>> v;
        for(int i=0;i<start.size();i++){
            v.push_back({start[i],end[i]});
        }
        
        sort(v.begin(),v.end(),compr);
        // Now we will make one stack and solve this
        // the size of maium stack will be its answer.
        
        int ans = 0;
        int endtime = -1;
        for(int i=0;i<v.size();i++){
            if(v[i].first > endtime){
                ans++;
                endtime = v[i].second;
            }
            
        }
        return ans;
    }
};


//{ Driver Code Starts.

int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        string input;
        int num;
        vector<int> start;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            start.push_back(num);
        }

        vector<int> end;
        getline(cin, input);
        stringstream s22(input);
        while (s22 >> num) {
            end.push_back(num);
        }

        Solution ob;
        int ans = ob.maxMeetings(start, end);
        cout << ans << endl;
        cout << "~" << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.py
#User function Template for python3

class Solution:
    
    #Function to find the maximum number of meetings that can
    #be performed in a meeting room.
    def maximumMeetings(self,start,end):
        # code here
        ans = []
        for i in range(0,len(start)):
            ans.append([start[i],end[i]])
        ans.sort(key=lambda x:x[1])
        
        i=0
        count =0
        while i<len(ans):
            j=i+1
            while j<len(ans) and ans[i][1]>=ans[j][0]:
                count+=1
                j+=1
            i=j
        return len(ans)-count